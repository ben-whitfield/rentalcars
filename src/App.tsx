import React, {useState, ChangeEvent} from 'react';
import style from './App.module.css';

import ListItem from './components/ListItem/ListItem';
import SearchForm from './components/SearchForm/SearchForm';
import SearchResults from './components/SearchResults/SearchResults';
import useFetchResults from './getResults';

function App() {
  const [ params, setParams ] = useState({searchTerm: ''})

  function handleParamChange(e: ChangeEvent<HTMLInputElement>) {
    const param = (e.target as HTMLInputElement).name
    const value = (e.target as HTMLInputElement).value
        
    setParams(prevParams => {
      return {...prevParams, [param]: value}
    })
  }
  
  const { results, loading, error } = useFetchResults(params.searchTerm)
  
  return (
      <div className={style['app']}>

        <section className={style['hero-container']}>
          <div className={style['hero-content']}>
            <h1 className={style['page-title']}>Car Hire â€“ Search, Compare & Save</h1>
            <ListItem text={'Free cancellations on most bookings'} icon={'check'} />
            <ListItem text={'60,000+ locations'} icon={'check'} />
            <ListItem text={'Customer support in 40+ languages'} icon={'check'} />
            <SearchForm params={params} onParamChange={handleParamChange}/>
            {<SearchResults results={results.results} loading={loading} error={error}/>}
          </div>
        </section>

        <section className={style['links']}>
          <h3>Page links</h3>
        </section>
      </div>
  );
}

export default App;
