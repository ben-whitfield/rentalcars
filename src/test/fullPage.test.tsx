import React from 'react';
import renderer from 'react-test-renderer';

import App from '../App';

describe('Snapshot', () => {
  test('snapshot render full page', () => {
    const component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});