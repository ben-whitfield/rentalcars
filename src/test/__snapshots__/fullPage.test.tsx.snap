// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot snapshot render full page 1`] = `
<div
  className="app"
>
  <section
    className="hero-container"
  >
    <div
      className="hero-content"
    >
      <h1
        className="page-title"
        role="heading"
      >
        Car Hire â€“ Search, Compare & Save
      </h1>
      <div
        className="list-container"
      >
        <div
          className="list-item"
        >
          <div
            className="list-icon-check"
          />
          <div
            className="list-text"
          >
            Free cancellations on most bookings
          </div>
        </div>
      </div>
      <div
        className="list-container"
      >
        <div
          className="list-item"
        >
          <div
            className="list-icon-check"
          />
          <div
            className="list-text"
          >
            60,000+ locations
          </div>
        </div>
      </div>
      <div
        className="list-container"
      >
        <div
          className="list-item"
        >
          <div
            className="list-icon-check"
          />
          <div
            className="list-text"
          >
            Customer support in 40+ languages
          </div>
        </div>
      </div>
      <div
        className="search-wrapper"
      >
        <form
          className="form"
        >
          <div
            className="search-input-wrapper"
          >
            <label
              className="search-label"
              htmlFor="searchform"
            >
              Search: 
            </label>
            <span
              className="search-icon"
            />
            <input
              className="search-input"
              data-testid="formInput"
              id="searchform"
              name="searchTerm"
              onChange={[Function]}
              placeholder="Pick-up location"
              type="text"
              value=""
            />
          </div>
          <button
            className="button"
            data-testid="searchButton"
            type="submit"
          >
            Search
          </button>
        </form>
      </div>
      <div
        className="search-results loading"
        data-testid="searchResultPanel"
      />
    </div>
  </section>
  <section
    className="links"
  >
    <h3>
      Page links
    </h3>
  </section>
</div>
`;
